test/unit/lib/challenge.test.ts

```diff
@@ -26,3 +27,173 @@ describe('Challenge Generation', (): void => {
     expect(Buffer.byteLength(challenge, 'base64')).toBe(8)
   })
 })
+
+// Each test generates a random key pair
+describe('Signature Verification', (): void => {
+  let challenge: string
+  let signer: crypto.Signer
+
+  beforeEach((): void => {
+    challenge = 'Crypto Auth service Yay!'
+    signer = crypto.createSign('SHA256')
+
+    signer.update(challenge)
+  })
+
+  it('verifies correct signature - EC Key', (): void => {
+    const keyPair = crypto.generateKeyPairSync('ec', {
+      namedCurve: 'secp256k1', // Allowed by FIDO spec
+      publicKeyEncoding: {
+        type: 'spki', // Key infrasructure
+        format: 'pem' // Encoding format
+      },
+      privateKeyEncoding: {
+        type: 'pkcs8',
+        format: 'pem'
+      }
+    })
+
+    const sign = signer.sign(keyPair.privateKey, 'base64')
+
+    expect(verifySign(challenge, sign, keyPair.publicKey)).toEqual(true)
+  })
```

Comment:
here you should check that verifier's `update` and `verify` was used properly,

Reply:
Not sure I agree. I think we want to test the behavioral end result not the implementation details of calling specific methods. Let's leave this to simply verify that the challenge checks out as we know it should.

Reply:
`verifySign` is returning `true` or `false` for a given set of parameters. We are not implementing crypto algorithm to verify the sign - we are delegating this difficult task to the `crypto` module, so we should test how we are using `crypto` interface.